framework/sandstone.cpp

```diff
@@ -1784,8 +1784,20 @@ static TestResult child_run(/*nonconst*/ struct test *test, int child_number)
         if (sApp->shmem->use_strict_runtime && wallclock_deadline_has_expired(sApp->endtime)){
             // skip cleanup on the last test when using strict runtime
         } else {
-            if (test->test_cleanup)
-                test->test_cleanup(test);
+            if (test->test_cleanup) {
+                ret = test->test_cleanup(test);
+                if (state == TestResult::Passed) {
+                    if (ret == EXIT_SKIP) {
+                        log_skip(RuntimeSkipCategory, "Skip requested in cleanup");
+                        state = TestResult::Skipped;
+                    } else if (ret < EXIT_SUCCESS) {
+                        log_skip(RuntimeSkipCategory, "Unexpected OS error in cleanup: %s", strerror(-ret));
+                        state = TestResult::Skipped;
```

Comment:
This is a skip, but it shouldn't say "Unexpected OS error".

Reply:
This is special case, identical with existing per init() implementation.

Reply:
Ok, I think this is acceptable though.